examples/calculations/submit_geomopt.py

```diff
@@ -0,0 +1,186 @@
+"""Example code for submitting single point calculation"""
+
+from pathlib import Path
+from typing import Union
+
+from ase.build import bulk
+from ase.io import read
+import click
+
+from aiida.common import NotExistent
+from aiida.engine import run_get_node
+from aiida.orm import Bool, Float, Str, StructureData, load_code, load_node
+from aiida.plugins import CalculationFactory
+
+from aiida_mlip.data.model import ModelData
+
+
+def load_model(string: Union[str, Path, None], architecture: str) -> ModelData:
+    """
+    Load a model from a given string.
+
+    If the string represents a file path, the model will be loaded from that path.
+    Otherwise, the model will be downloaded from the specified location.
+
+    Parameters
+    ----------
+    string : Union[str, Path, None]
+        The string representing either a file path or a URL for downloading the model.
+    architecture : str
+        The architecture of the model.
+
+    Returns
+    -------
+    ModelData or None
+        The loaded model if successful, otherwise None.
+    """
+    if string is None:
+        model = None
+    elif (file_path := Path(string)).is_file():
+        model = ModelData.local_file(file_path, architecture=architecture)
+    else:
+        model = ModelData.download(string, architecture=architecture)
+    return model
+
+
+def load_structure(struct: Union[str, Path, int, None]) -> StructureData:
+    """
+    Load a StructureData instance from the given input.
+
+    The input can be either a path to a structure file, a node PK (int),
+    or None. If the input is None, a default StructureData instance for NaCl
+    with a rocksalt structure will be created.
+
+    Parameters
+    ----------
+    struct : Union[str, Path, int, None]
+        The input value representing either a path to a structure file, a node PK,
+        or None.
+
+    Returns
+    -------
+    StructureData
+        The loaded or created StructureData instance.
+
+    Raises
+    ------
+    click.BadParameter
+        If the input is not a valid path to a structure file or a node PK.
+    """
+    if struct is None:
+        structure = StructureData(ase=bulk("NaCl", "rocksalt", 5.63))
+    elif isinstance(struct, int) or (isinstance(struct, str) and struct.isdigit()):
+        structure_pk = int(struct)
+        structure = load_node(structure_pk)
+    elif Path.exists(struct):
+        structure = StructureData(ase=read(struct))
+    else:
+        raise click.BadParameter(
+            f"Invalid input: {struct}. Must be either node PK (int) or a valid \
+                path to a structure file."
+        )
+    return structure
+
+
+def singlepoint(params: dict) -> None:
+    """
+    Prepare inputs and run a single point calculation.
+
+    Parameters
+    ----------
+    params : dict
+        A dictionary containing the input parameters for the calculations
+
+    Returns
+    -------
+        None
+    """
+    for key, value in params.items():
```

Comment:
do we need this print?

Reply:
not really, just debugging things I forget to remove